src/main/java/de/teamlapen/vampirism/command/LevelCommand.java

```diff
@@ -47,15 +54,23 @@ private static int setLevel(CommandContext<CommandSource> context, IPlayableFact
         for (ServerPlayerEntity player : players) {
             FactionPlayerHandler handler = FactionPlayerHandler.get(player);
             if (level == 0 && !handler.canLeaveFaction()) {
-                context.getSource().sendErrorMessage(new TranslationTextComponent("command.vampirism.base.level.cant_leave"));
+                context.getSource().sendErrorMessage(new TranslationTextComponent("command.vampirism.base.level.cant_leave", players.size() > 1 ? player.getDisplayName() : "Player", handler.getCurrentFaction().getName()));
             }
             if (handler.setFactionAndLevel(faction, level)) {
                 context.getSource().sendFeedback(new TranslationTextComponent("command.vampirism.base.level.successful", player.getName(), faction.getName(), level), true);
             } else {
-                context.getSource().sendErrorMessage(new TranslationTextComponent("command.vampirism.failed_to_execute"));
+                context.getSource().sendErrorMessage(players.size() > 1 ? new TranslationTextComponent("command.vampirism.failed_to_execute.players", player.getDisplayName()) : new TranslationTextComponent("command.vampirism.failed_to_execute"));
             }
         }
         return 0;
     }
 
+    private static int noFaction(Collection<ServerPlayerEntity> players) {
```

Comment:
The name of this method isn't clear at all. And why does it return 0?

Reply:
Also why does the setLevel return 0

Reply:
the execute() method needs a lambda with int return type, but as far as i know the value is irrelevant.

Reply:
ok